### PR TITLE
Add support for STM32L0 series timers and PWM driver

### PR DESCRIPTION
**Pull request by martinjaeger**
This PR adds all timers and PWM channels for the STM32L0 series MCUs to devicetree and updates the PWM driver to support also the L0 series.

I was not sure what the rule for setting the the PWM prescaler in the devicetree is. For other MCUs it's either set to `0` or to `10000`, but I could not see a pattern. As the L0 MCUs have rather low system clock frequency, I chose `0` to allow high PWM signal frequencies by default.

Tested on `nucleo_l073rz`.

### CODE REVIEW

File: drivers/pwm/pwm_stm32.c [hunk 1]
```diff
@@ -313,16 +318,21 @@ static int pwm_stm32_init(const struct device *dev)
 	init.CounterMode = LL_TIM_COUNTERMODE_UP;
 	init.Autoreload = 0u;
 	init.ClockDivision = LL_TIM_CLOCKDIVISION_DIV1;
+#ifndef CONFIG_SOC_SERIES_STM32L0X
```
Comment (gmarull) [inline drivers/pwm/pwm_stm32.c:321]: I think `RepetitionCounter` could actually be removed, `LL_TIM_StructInit` takes care of initializing the field if it's available to zero.

File: drivers/pwm/pwm_stm32.c [hunk 2]
```diff
@@ -233,7 +238,9 @@ static int pwm_stm32_pin_set(const struct device *dev, uint32_t pwm,
 		oc_init.OCState = LL_TIM_OCSTATE_ENABLE;
 		oc_init.CompareValue = pulse_cycles;
 		oc_init.OCPolarity = get_polarity(flags);
+#ifndef CONFIG_SOC_SERIES_STM32L0X
```
Comment (gmarull) [inline drivers/pwm/pwm_stm32.c:241]: Could probably be removed as it is initialized by `LL_TIM_OC_StructInit` if available.

File: drivers/pwm/pwm_stm32.c (hunk 1 above)
Comment (martinjaeger) [reply drivers/pwm/pwm_stm32.c]: You are totally correct. Double-checked the init function for all MCUs and removed it.

File: drivers/pwm/pwm_stm32.c (hunk 2 above)
Comment (martinjaeger) [reply drivers/pwm/pwm_stm32.c]: Correct also here. Double-checked the init function for all MCUs. It is safe to be removed.

**Review by gmarull — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**